cdk/domino_cdk/eks_stack.py

```diff
@@ -400,17 +402,78 @@ def provision_eks_iam_policies(self):
             managed_policies=managed_policies,
         )
 
+    def _get_machine_image(self, cfg_name: str, cfg: Dict[str, Any]) -> Tuple[Optional[str], Optional[str]]:
+        ami_id = cfg.get("ami_id")
+        user_data = cfg.get("user_data")
+        if ami_id and not user_data:
+            raise Exception("User data must be provided when specifying a custom AMI")
+        if ami_id and (cfg.get("ssm_agent") or cfg.get("labels") or cfg.get("taints")):
+            raise Exception(
+                "ssm_agent, labels and taints cannot be automatically configured when specifying a custom AMI. "
+                "You need to configure all of this using user_data."
+            )
+        return ami_id, user_data
+
+    def _handle_user_data(
+        self, name: str, custom_ami: bool, cfg: dict, user_data_list: List[Union[ec2.UserData, str]]
+    ) -> Optional[ec2.UserData]:
+        proceed: bool = False
+        mime_user_data = ec2.MultipartUserData()
+
+        # If we are using default EKS image, tweak kubelet
+        if not custom_ami:
+            mime_user_data.add_part(
+                ec2.MultipartBody.from_user_data(
+                    ec2.UserData.custom(
+                        'KUBELET_CONFIG=/etc/kubernetes/kubelet/kubelet-config.json\n'
+                        'echo "$(jq \'.eventRecordQPS=0\' $KUBELET_CONFIG)" > $KUBELET_CONFIG'
+                    )
+                ),
+            )
+            proceed = True
+        if cfg.get("ssm_agent"):
```

Comment:
Since we only do this with a custom ami, I wonder if we should indent this once? It's not really _needed_ because of the exception elsewhere, but organizationally I think it makes sense, as reading the code it might be more obvious what the intent is.

Reply:
Agree. Per our conversation 2 days ago, we should only yum if it is not a custom ami.